### PR TITLE
Lab28gavin

### PR DESCRIPTION
**Pull request by GavinThomas1192**

### CODE REVIEW

File: labgavin/src/component/category-item/_category-item.scss [hunk 1]
```diff
@@ -15,7 +15,30 @@ body {
     background: grey;
     width: 100%;
   }
-  .expenseContainer {
-    display: inline;
+  .expense-content {
```
Comment (izzybaer) [inline labgavin/src/component/category-item/_category-item.scss]: good job!

File: labgavin/src/component/category-item/_category-item.scss [hunk 2]
```diff
@@ -15,7 +15,30 @@ body {
...
+  .expense-items {
+    display: inline-block;
+  }
+  .expense-items p {
+    max-width: 100%;
+    display: inline-block;
+    margin: 0px 10px 0px 10px;
   }
 }
+// button, select, option {
```
Comment (izzybaer) [inline labgavin/src/component/category-item/_category-item.scss]: corpse code

File: labgavin/src/component/category-item/_category-item.scss [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  .expense-items {
+    display: inline-block;
+  }
+  .expense-items p {
+    max-width: 100%;
+    display: inline-block;
+    margin: 0px 10px 0px 10px;
+  }
+}
+// button, select, option {
```
Comment (izzybaer) [inline labgavin/src/component/category-item/_category-item.scss:32]: corpse

File: labgavin/src/component/category-item/index.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      editCategory: false,
+      expenseBox: false,
+    };
+    this.toggleCategory = this.toggleCategory.bind(this);
+    this.toggleExpense = this.toggleExpense.bind(this);
+  }
+
+  toggleCategory() {
+    this.setState({editCategory: !this.state.editCategory});
+    console.log('thi');
```
Comment (izzybaer) [inline labgavin/src/component/category-item/index.js:24]: remove this

File: labgavin/src/component/dashboard-container/_dashboard-container.scss [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+$gutter-lg: $gutter-sm * 2;
+$font-primary: 'helvetica';
+$border-radius: 3px;
+
+body {
+  background: $secondary;
+  font-family: $font-primary;
+  text-align: center;
+
+  .categoryForm {
```
Comment (izzybaer) [inline labgavin/src/component/dashboard-container/_dashboard-container.scss:15]: this should be kabob-case

File: labgavin/src/component/dashboard-container/_dashboard-container.scss [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  background: $white;
+  border: 1px solid $black;
+  transition: 350ms;
+
+  &:hover {
+    background: $primary;
+  }
+}
+
+.myP {
```
Comment (izzybaer) [inline labgavin/src/component/dashboard-container/_dashboard-container.scss:39]: kabob-case

File: labgavin/src/component/dashboard-container/index.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+      <main className="dashboard-container">
+        <h1>Track your Budget</h1>
+        <h2>Total Monthly Budget: {this.props.totalMonthly}</h2>
+        <h2>Total remaining: {this.props.totalRemaining}</h2>
+        <div className='categoryForm'>
+          <CategoryForm
+            buttonText='Create'
+            onComplete={this.props.categoryCreate} />
+        </div>
+        <div className="categoryList">
```
Comment (izzybaer) [inline labgavin/src/component/dashboard-container/index.js:29]: kabob-case

File: labgavin/src/component/dashboard-container/index.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+  }
+}
+
+
+
+const mapStateToProps = state => {
+  return {
+    categories: state.categories,
+    totalMonthly: 2000,
+    // totalRemaining: 2000 - this.state.totalSpent,
```
Comment (izzybaer) [inline labgavin/src/component/dashboard-container/index.js:52]: corpse

File: labgavin/src/component/expense-form/index.js [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+  render() {
+    return (
+      <form className='expenseForm' onSubmit={this.handleSubmit}>
+        <h4>{this.props.buttonText} expense</h4>
+        <input
+          name='title'
+          type='text'
+          placeholder='title'
+          value={this.state.value}
```
Comment (izzybaer) [inline labgavin/src/component/expense-form/index.js:44]: shouldn't this be this.state.title?

File: labgavin/src/component/expense-form/index.js [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+          onChange={this.handleChange} /><br/>
+        <input
+          name='price'
+          type='number'
+          placeholder='price'
+          value={this.state.value}
```
Comment (izzybaer) [inline labgavin/src/component/expense-form/index.js:50]: and this.state.price?

File: labgavin/src/component/dashboard-container/index.js (hunk 7 above)
Comment (GavinThomas1192) [reply labgavin/src/component/dashboard-container/index.js:29]: No, I'm allowed to write code how I like.

File: labgavin/src/lib/validation-middleware.js [hunk 11]
```diff
@@ -0,0 +1,42 @@
+// let Validate = store => next => action => {
```
Comment (izzybaer) [inline labgavin/src/lib/validation-middleware.js:1]: so.much.corpse.code

File: labgavin/src/component/dashboard-container/index.js (hunk 7 above)
Comment (GavinThomas1192) [reply labgavin/src/component/dashboard-container/index.js:29]: Jk, Izzy is always right.

File: labgavin/src/reducer/expenses.js [hunk 12]
```diff
@@ -0,0 +1,52 @@
+let initialState = {};
+
+let validateCategory = (category) => {
+  if (!category.id || !category.title || !category.timestamp || !category.budget) {
+    throw new Error('VALIDATION ERROR: category must include id, title, timestamp and budget');
+  }
+};
+
+let validateExpense = (expense) => {
+  console.log(expense,'yoyoyooy');
```
Comment (izzybaer) [inline labgavin/src/reducer/expenses.js:10]: remove this

File: labgavin/src/style/main.scss [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+  background: $white;
+  border: 1px solid $black;
+  transition: 350ms;
+
+  &:hover {
+    background: $primary;
+  }
+}
+
+.myP {
```
Comment (izzybaer) [inline labgavin/src/style/main.scss:50]: kabob-case

**Pull request merged**